Common/cpp/NativeReanimatedModule.cpp

```diff
@@ -206,6 +206,51 @@ void NativeReanimatedModule::onEvent(std::string eventName, std::string eventAsS
   applierRegistry->event(*runtime, eventName, ho);
 }
 
+void NativeReanimatedModule::getRegistersState(jsi::Runtime &rt, int option, const jsi::Value &value) {
+  // option:
+  //  1 - shared values
+  //  2 - worklets
+  //  3 - appliers
+  jsi::Function fun = value.getObject(rt).asFunction(rt);
+  std::shared_ptr<jsi::Function> funPtr(new jsi::Function(std::move(fun)));
+
+  scheduler->scheduleOnUI([&rt, funPtr, this, option]() {
+    std::string ids;
+    switch(option) {
+      case 1: {
+        for(auto it = sharedValueRegistry->sharedValueMap.begin(); it != sharedValueRegistry->sharedValueMap.end(); ++it) {
```

Comment:
Can you use foreach loop?

Reply:
fixed 70dd9c2b456dc5edb71581e9c70aded6a060f804